student-manager/src/main/java/spring/studentmanager/controller/StudentController.java

```diff
@@ -30,17 +30,11 @@
 		Student foundStudent = students.stream()
 				.filter(s->s.getId()==studentId)
 				.findFirst().orElse(null);
-		if(foundStudent != null) {
-			return new ResponseEntity<>(foundStudent, HttpStatus.OK);
-		}
-		return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+		return new ResponseEntity<>(foundStudent, HttpStatus.OK);
```

Comment:
You should not have to remove 404 from this method, In this method you have to return 404 if no student is found. You had to remove 404 from findAll() only.